homeassistant/components/kmtronic/switch.py

```diff
@@ -0,0 +1,63 @@
+"""KMtronic Switch integration."""
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DATA_COORDINATOR, DATA_HOST, DATA_HUB, DOMAIN
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Config entry example."""
+    coordinator = hass.data[DOMAIN][entry.entry_id][DATA_COORDINATOR]
+    hub = hass.data[DOMAIN][entry.entry_id][DATA_HUB]
+    host = hass.data[DOMAIN][entry.entry_id][DATA_HOST]
+    await hub.async_get_relays()
+
+    async_add_entities(
+        [KMtronicSwitch(coordinator, host, relay) for relay in hub.relays]
+    )
+
+
+class KMtronicSwitch(CoordinatorEntity, SwitchEntity):
+    """KMtronic Switch Entity."""
+
+    def __init__(self, coordinator, host, relay):
+        """Pass coordinator to CoordinatorEntity."""
+        super().__init__(coordinator)
+        self._host = host
+        self._relay = relay
+
+    @property
+    def available(self) -> bool:
+        """Return whether the entity is available."""
+        return self.coordinator.last_update_success
+
+    @property
+    def name(self) -> str:
+        """Return the name of the entity."""
+        return f"Relay{self._relay.id}"
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique ID of the entity."""
+        return f"{self._host}_relay{self._relay.id}"
```

Comment:
```suggestion
        return f"{self._config_entry_id}_relay{self._relay.id}"
```

Let's tie it to the config entry instead.  If they change the host by reauth later or some other means it won't affect the entity.